sparse/core.py

```diff
@@ -611,34 +619,46 @@ def elemwise_binary(self, func, other, *args, **kwargs):
 
         # Sort self.coords in lexographical order using record arrays
         self_coords = np.rec.fromarrays(self.coords)
-        i = np.argsort(self_coords)
-        self_coords = self_coords[i]
-        self_data = self.data[i]
+        self_sort_order = np.argsort(self_coords)
+        self_coords = self_coords[self_sort_order]
+        self_data = self.data[self_sort_order]
 
-        # Convert other.coords to a record array
+        # Sort other.coords in lexographical order using record arrays
         other_coords = np.rec.fromarrays(other.coords)
-        other_data = other.data
+        other_sort_order = np.argsort(other_coords)
+        other_coords = other_coords[other_sort_order]
+        other_data = other.data[other_sort_order]
 
         # Find matches between self.coords and other.coords
-        j = np.searchsorted(self_coords, other_coords)
-        if len(self_coords):
-            matched_other = (other_coords == self_coords[j % len(self_coords)])
-        else:
-            matched_other = np.zeros(shape=(0,), dtype=bool)
-        matched_self = j[matched_other]
+        # TODO: implement a modified exponential search to find next fitting coordinate faster
+        matched_self = np.zeros(shape=(len(self_coords),), dtype=bool)
+        matched_other = np.zeros(shape=(len(other_coords),), dtype=bool)
+        i = 0
+        j = 0
+        while (i < len(self_coords)) & (j < len(other_coords)):
+            self_coord = tuple(self_coords[i])
+            other_coord = tuple(other_coords[j])
+            if self_coord == other_coord:
+                matched_self[i] = True
+                matched_other[j] = True
+                i += 1
+                j += 1
+            elif self_coord < other_coord:
+                i += 1
+            else:
+                j += 1
```

Comment:
What is the performance impact of this?  I would expect iterating through a numpy array and creating many tuples to be somewhat expensive.

Reply:
This sounds like an asymptotic argument.  Have you actually benchmarked and tested the speed difference here?

What is this change supposed to do?  Is it necessary to handle the elemwise and and or operators?

Reply:
Of course it is an asymptotic argument and I didn't benchmark it. Sorry, I didn't have time for that then.

Anyways the change from line 634 to 649 is not to handle elemwise AND and OR operations but to handle any elemwise operations correctly.

If you take a look at the example from #25.

It takes two tensors - one empty and one with a True at (0, 1, 2) - and calculates the elemwise OR. The result is an empty tensor. But it should be a tensor with an True or 1 at (0, 1, 2).

You get the same result if you take equivalent int or float valued input tensors and ADD as operation. Result is again an empty (all 0) tensor. So there is definitely a bug in the coords matching between the two input tensors.

I know this is not the most beautiful solution but at least it returns correct results for now.